exercises/practice/rational-numbers/test_rational_numbers.zig

```diff
@@ -0,0 +1,221 @@
+    try testing.expectEqual(expected, actual);
+}
+
+test "divide a rational number by 1" {
+    const expected = try numbers.Rational.init(1, 2);
+    const actual = (numbers.Rational.init(1, 2) catch unreachable).div(numbers.Rational.init(1, 1) catch unreachable);
+    try testing.expectEqual(expected, actual);
+}
+
+test "absolute value of a positive rational number" {
+    const expected = try numbers.Rational.init(1, 2);
+    const actual = (numbers.Rational.init(1, 2) catch unreachable).abs();
+    try testing.expectEqual(expected, actual);
+}
+
+test "absolute value of a negative rational number" {
+    const expected = try numbers.Rational.init(1, 2);
+    const actual = (numbers.Rational.init(-1, 2) catch unreachable).abs();
+    try testing.expectEqual(expected, actual);
+}
+
+test "absolute value of zero" {
+    const expected = try numbers.Rational.init(0, 1);
+    const actual = (numbers.Rational.init(0, 1) catch unreachable).abs();
+    try testing.expectEqual(expected, actual);
+}
+
+test "raise a positive rational number to a positive integer power" {
+    const expected = try numbers.Rational.init(1, 8);
+    const actual = (numbers.Rational.init(1, 2) catch unreachable).exp(3);
+    try testing.expectEqual(expected, actual);
+}
+
+test "raise a negative rational number to a positive integer power" {
+    const expected = try numbers.Rational.init(-1, 8);
+    const actual = (numbers.Rational.init(-1, 2) catch unreachable).exp(3);
+    try testing.expectEqual(expected, actual);
+}
+
+test "raise zero to an integer power" {
+    const expected = try numbers.Rational.init(0, 1);
+    const actual = (numbers.Rational.init(0, 1) catch unreachable).exp(5);
+    try testing.expectEqual(expected, actual);
+}
+
+test "raise one to an integer power" {
+    const expected = try numbers.Rational.init(1, 1);
+    const actual = (numbers.Rational.init(1, 1) catch unreachable).exp(4);
+    try testing.expectEqual(expected, actual);
+}
+
+test "raise a positive rational number to the power of zero" {
+    const expected = try numbers.Rational.init(1, 1);
+    const actual = (numbers.Rational.init(1, 2) catch unreachable).exp(0);
+    try testing.expectEqual(expected, actual);
+}
+
+test "raise a negative rational number to the power of zero" {
+    const expected = try numbers.Rational.init(1, 1);
+    const actual = (numbers.Rational.init(-1, 2) catch unreachable).exp(0);
+    try testing.expectEqual(expected, actual);
+}
+
+test "raise a real number to a positive rational number" {
+    const expected = 16.0;
+    const actual = comptime (numbers.Rational.init(4, 3) catch unreachable).exp_real(8);
+    try testing.expectEqual(expected, actual);
+}
+
+test "raise a real number to a negative rational number" {
+    const tolerance = 0.001;
+    const expected = 0.333;
+    const actual = comptime (numbers.Rational.init(-1, 2) catch unreachable).exp_real(9);
+    try testing.expect(std.math.approxEqAbs(f64, expected, actual, tolerance));
```

Comment:
Had to use `std.math.approxEqAbs` instead of `testing.expectApproxEqAbs` since the latter cannot do a `comptime` check 🧇 😲 .